engine/classes/ElggUser.php

```diff
@@ -127,6 +127,8 @@ public function ban($reason = '') {
 		if (!_elgg_services()->events->trigger('ban', 'user', $this)) {
 			return false;
 		}
+		
+		_elgg_services()->events->trigger('disable:after', 'all', $this);
```

Comment:
this is a completely wrong event at this time. nothing is getting disabled.
The correct way would be to trigger an after event after the user is actually banned in the DB

Reply:
please check the updated commit